subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptorTest.groovy

```diff
@@ -242,14 +256,17 @@ class DefaultRepositoryContentDescriptorTest extends Specification {
         }
 
         where:
-        method         | expr     | group   | module | version | excluded
-        'Group'        | 'org'    | 'org'   | 'foo'  | '1.0'   | true
-        'Group'        | 'org'    | 'other' | 'foo'  | '1.0'   | false
-        'GroupByRegex' | 'org'    | 'org'   | 'foo'  | '1.0'   | true
-        'GroupByRegex' | 'org'    | 'other' | 'foo'  | '1.0'   | false
-        'GroupByRegex' | 'bar'    | 'org'   | 'foo'  | '1.0'   | false
-        'GroupByRegex' | '[org]+' | 'org'   | 'foo'  | '1.0'   | true
-        'GroupByRegex' | '[org]+' | 'other' | 'foo'  | '1.0'   | false
+        method              | expr     | group        | module | version | excluded
+        'Group'             | 'org'    | 'org'        | 'foo'  | '1.0'   | true
+        'Group'             | 'org'    | 'other'      | 'foo'  | '1.0'   | false
+        'GroupByRegex'      | 'org'    | 'org'        | 'foo'  | '1.0'   | true
+        'GroupByRegex'      | 'org'    | 'other'      | 'foo'  | '1.0'   | false
+        'GroupByRegex'      | 'bar'    | 'org'        | 'foo'  | '1.0'   | false
+        'GroupByRegex'      | '[org]+' | 'org'        | 'foo'  | '1.0'   | true
+        'GroupByRegex'      | '[org]+' | 'other'      | 'foo'  | '1.0'   | false
+        'GroupAndSubGroups' | 'org'    | 'org'        | 'foo'  | '1.0'   | true
+        'GroupAndSubGroups' | 'org'    | 'org.other'  | 'foo'  | '1.0'   | true
+        'GroupAndSubGroups' | 'org'    | 'org1.other' | 'foo'  | '1.0'   | false
```

Comment:
🤔 Some creative additional negative test cases to help fully define the behavior of the new method could be useful.  Things like:

```
method 				| expr 		| group 			| module 	| version
'GroupAndSubGroups' | 'org' 	| 'ORG.other'	 	| 'foo' 	| '1.0'
'GroupAndSubGroups' | 'org' 	| 'org.other.org' 	| 'foo' 	| '1.0'
'GroupAndSubGroups' | 'org.' 	| 'org.other'	 	| 'foo' 	| '1.0'
'GroupAndSubGroups' | 'org' 	| 'org.org.other' 	| 'foo' 	| '1.0'
'GroupAndSubGroups' | '.' 		| 'org.other'	 	| 'foo' 	| '1.0'
'GroupAndSubGroups' | 'nope' 	| 'org.other'	 	| 'foo' 	| '1.0'